### PR TITLE
 Include key type and scheme in PublicKey's PartialEq and Hash

### PR DESCRIPTION
**Pull request by erickt**
Now that we've switched over to using the TUF 1.0 scheme for keyids, the public key's value no longer incorporates the key's type and signature scheme. This fixes things so that equality and hashing work correctly with different key type and signature schemes, but the same public key bytes.

### CODE REVIEW

File: src/crypto.rs [hunk 1]
```diff
@@ -595,8 +595,7 @@ impl PublicKey {
 
 impl PartialEq for PublicKey {
     fn eq(&self, other: &Self) -> bool {
-        // the other two fields are derived from this one, we ignore them
```
Comment (ComputerDruid) [inline src/crypto.rs:598]: Maybe leave this comment explaining that key_id is computed from these 3?

File: src/interchange/cjson/shims.rs [hunk 2]
```diff
@@ -381,7 +381,9 @@ impl TargetDescription {
         TargetDescription {
             length: description.length(),
             hashes: description.hashes().iter().map(|(k, v)| (k.clone(), v.clone())).collect(),
-            custom: description.custom().map(|custom| custom.iter().map(|(k, v)| (k.clone(), v.clone())).collect()),
+            custom: description
+                .custom()
+                .map(|custom| custom.iter().map(|(k, v)| (k.clone(), v.clone())).collect()),
```
Comment (ComputerDruid) [inline src/interchange/cjson/shims.rs]: this looks unrelated to this change.

File: src/crypto.rs [hunk 3]
```diff
@@ -616,7 +615,8 @@ impl PartialOrd for PublicKey {
 
 impl hash::Hash for PublicKey {
     fn hash<H: hash::Hasher>(&self, state: &mut H) {
-        // the other two fields are derived from this one, we ignore them
```
Comment (ComputerDruid) [inline src/crypto.rs:619]: ditto about ignoring keyid

**Review by wellsie1116 — APPROVED** [review]
